tendermint/src/block/header.rs

```diff
@@ -139,10 +142,20 @@ pub struct Version {
     pub block: u64,
 
     /// App version
-    #[serde(with = "serializers::from_str")]
+    #[serde(
+        with = "serializers::from_str",
+        default = "Version::default_app_version"
```

Comment:
Do we really need our own function here? Can't we specify something like "use the default for this type" which would be 0?

Reply:
cc @greg-szabo

Reply:
I did this consciously to be explicit about the default value. The following has the same effect, but only coincidentally:

```rust
#[derive(Serialize, Deserialize, Clone, PartialEq, Debug)]
pub struct Version {
    /// Block version
    #[serde(with = "serializers::from_str")]
    pub block: u64,

    /// App version
    #[serde(with = "serializers::from_str", default)]
    pub app: u64,
}
```

Reply:
I think it's still clear if `#[serde(default)]` is used here.